src/main/java/org/opengis/cite/geomatics/gml/GmlUtils.java

```diff
@@ -546,37 +546,48 @@ public static Element convertToMultiType(Node geomNode) {
         Element memberType = document.createElementNS(geomNode.getNamespaceURI(), geomMemberType);
         Node importNode = document.importNode(geomNode, true);
         NamedNodeMap attributes = geomNode.getAttributes();
+        String srsName = null;
 
         for (Integer i = 0; i < attributes.getLength(); i++) {
             String attributeNamespace = attributes.item(i).getNamespaceURI();
             String attributeName = attributes.item(i).getLocalName();
             String attributeValue = attributes.item(i).getNodeValue();
 
             multiGeom.setAttributeNS(attributeNamespace, attributeName, attributeValue);
+            if (attributeName.equalsIgnoreCase("srsName")) {
+                srsName = GeodesyUtils.convertSRSNameToURN(attributeValue);
+            }
         }
 
-        // In case of Surface geometry type
-        if (typeName.equalsIgnoreCase("MultiSurface")) {
```

Comment:
@keshav-nangare 
Why did you remove the handling of "MultiSurface"? Shouldn't this be treated as an additional case?

Reply:
The "MultiSurface" case will be handled in the else condition.

Reply:
The structure for the "MultiCurve" is: 
```
<MultiCurve>
  <curveMember>
    <LineString srsName=>
      <posList></posList>
    </LineString>
  </curveMember>
</MultiCurve>
```

The structure for the "MultiSuface" is: 
```
<MultiSuface>
  <surfaceMember>
    <Polygon srsName=>
      <exterior></exterior>
    </Polygon>
  </surfaceMember>
</MultiSuface>
```

So if condition will check for the "MultiCurve" to form the valid structure and else condition for the "MultiSurface".